apps/preview/next/pages/showcases/MegaMenu/OpenOnClick.tsx

```diff
@@ -0,0 +1,263 @@
+                        <h2
+                          role="presentation"
+                          className="typography-text-xs font-bold text-neutral-900 tracking-widest whitespace-nowrap px-4 py-2"
+                        >
+                          {heading}
+                        </h2>
+                        <hr className="mb-3.5" />
+                        <ul>
+                          {items.map((item) => (
+                            <SfListItem size="sm" role="none" key={item.title}>
+                              <a
+                                className="focus-visible:outline focus-visible:outline-offset focus-visible:rounded-sm"
+                                href={item.link}
+                              >
+                                {item.title}
+                              </a>
+                            </SfListItem>
+                          ))}
+                        </ul>
+                      </div>
+                    ))}
+                    <div className="flex flex-col items-center justify-center bg-neutral-100 rounded-md border-neutral-300 overflow-hidden grow">
+                      <img src={bannerDetails.image.src} alt={bannerDetails.title} className="object-contain" />
+                      <p className="mb-4 mt-4 px-4 text-center typography-text-base font-medium">
+                        {bannerDetails.title}
+                      </p>
+                    </div>
+                    <SfButton
+                      square
+                      size="sm"
+                      variant="tertiary"
+                      aria-label="Close drawer"
+                      onClick={() => {
+                        setOpen(!open);
+                      }}
+                      className="hidden md:block md:absolute md:right-0 hover:bg-white active:bg-white"
+                    >
+                      <SfIconClose className="text-neutral-500" />
+                    </SfButton>
+                  </SfDrawer>
+                </CSSTransition>
+              </li>
+            </ul>
+            <div>
+              {actionItems.map((actionItem) => (
+                <SfButton
+                  className={classNames(
+                    'mr-2 -ml-0.5 text-white bg-transparent hover:bg-primary-800 hover:text-white active:bg-primary-900 active:text-white',
+                    actionItem.label === 'Log in' ? 'md:hidden' : null,
+                  )}
+                  key={actionItem.ariaLabel}
+                  aria-label={actionItem.ariaLabel}
+                  variant="tertiary"
+                  slotPrefix={actionItem.icon}
+                  square
+                />
+              ))}
+              <SfButton
+                className="hidden md:inline-flex mr-2 -ml-0.5 text-white bg-transparent hover:bg-primary-800 hover:text-white active:bg-primary-900 active:text-white"
+                key={actionItems[2].ariaLabel}
```

Comment:
Depending on index number don't seem like a good idea. Users may and most likely change this array, so we must handle this special item in a different way. I've seen you already check it on line [236](https://github.com/vuestorefront/storefront-ui/pull/2577/files#diff-467fc5ab5e397566ae141f22613ae257c22a2e5e872839af4b62c81ac570bb11R236) `actionItem.label === 'Log in'`, so maybe it's worth to add some kind of an indicator like `isLogin` to this item, or event declare it separately.

Reply:
@dkacper sure, it was copied from the current NavbarTop so we should change it there also.